spec/system/articles/user_visits_articles_by_timeframe_spec.rb

```diff
@@ -15,185 +16,138 @@
     create(:article, :past, past_published_at: 2.years.ago, score: minimum_score, user: author)
   end
 
-  def shows_correct_articles_count(count)
-    expect(page).to have_selector(".crayons-story__title", visible: :visible, count: count)
+  before do
+    Timecop.freeze(Time.current)
+  end
+
+  after do
+    Timecop.return
   end
 
-  def shows_correct_articles_count_via_xpath(count)
-    expect(page).to have_xpath(
-      "//article[contains(@class, 'crayons-story') and not(contains(@class, 'crayons-story--featured'))]",
-      count: count,
-    )
+  def shows_correct_articles_count(count)
+    expect(page).to have_selector(".crayons-story", visible: :all, count: count)
```

Comment:
The bulk of the change is here. This should ensure a consistent result. `visibale: :all` because there's no way the test browser can show 5 articles without scrolling down.